firebase-firestore/src/test/java/com/google/firebase/firestore/FirebaseFirestoreTest.java

```diff
@@ -0,0 +1,154 @@
+import static org.junit.Assert.fail;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.emulators.EmulatedServiceSettings;
+import com.google.firebase.emulators.EmulatorSettings;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class FirebaseFirestoreTest {
+
+  @Test
+  public void getInstance_withEmulator() {
+    FirebaseApp app = getApp("getInstance_withEmulator");
+
+    app.enableEmulators(
+        new EmulatorSettings.Builder()
+            .addEmulatedService(
+                FirebaseFirestore.EMULATOR, new EmulatedServiceSettings("10.0.2.2", 8080))
+            .build());
+
+    FirebaseFirestore firestore = FirebaseFirestore.getInstance(app);
+    FirebaseFirestoreSettings settings = firestore.getFirestoreSettings();
+
+    assertEquals(settings.getHost(), "10.0.2.2:8080");
+    assertFalse(settings.isSslEnabled());
+  }
+
+  @Test
+  public void getInstance_withEmulator_mergeSettingsSuccess() {
+    FirebaseApp app = getApp("getInstance_withEmulator_mergeSettingsSuccess");
+    app.enableEmulators(
+        new EmulatorSettings.Builder()
+            .addEmulatedService(
+                FirebaseFirestore.EMULATOR, new EmulatedServiceSettings("10.0.2.2", 8080))
+            .build());
+
+    FirebaseFirestore firestore = FirebaseFirestore.getInstance(app);
+    firestore.setFirestoreSettings(
+        new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(false).build());
+
+    FirebaseFirestoreSettings settings = firestore.getFirestoreSettings();
+
+    assertEquals(settings.getHost(), "10.0.2.2:8080");
+    assertFalse(settings.isSslEnabled());
+    assertFalse(settings.isPersistenceEnabled());
+  }
+
+  @Test
+  public void getInstance_withEmulator_mergeSettingsFailure() {
+    FirebaseApp app = getApp("getInstance_withEmulator_mergeSettingsFailure");
+    app.enableEmulators(
+        new EmulatorSettings.Builder()
+            .addEmulatedService(
+                FirebaseFirestore.EMULATOR, new EmulatedServiceSettings("10.0.2.2", 8080))
+            .build());
+
+    try {
+      FirebaseFirestore firestore = FirebaseFirestore.getInstance(app);
+      firestore.setFirestoreSettings(
+          new FirebaseFirestoreSettings.Builder().setHost("myhost.com").build());
+      fail("Exception should be thrown");
+    } catch (Exception e) {
+      assertTrue(e instanceof IllegalStateException);
+      assertEquals(
+          e.getMessage(),
+          "Cannot specify the host in FirebaseFirestoreSettings when EmulatedServiceSettings is provided.");
+    }
+  }
+
+  @Test
+  public void setSettings_repeatedSuccess() {
+    FirebaseApp app = getApp("setSettings_repeatedSuccess");
```

Comment:
The problem in the original PR would have only manifested itself if EmulatorSettings are also provided, which would have made the local settings different from the user's. Can you add the preamble from the other tests do to invoke this behavior?

Reply:
Added one more test for that.